spec/services/kubernetes_adapter_spec.rb

```diff
@@ -85,41 +73,103 @@
         allow(ENV).to receive(:[]).with('RUNNER_SENTRY_DSN')
                                   .and_return('runner-sentry-dsn-here')
 
-        subject.create_deployment(config_dir: config_dir,
-                                  name: nil,
-                                  container_port: nil,
-                                  image: nil,
-                                  json_repo: nil,
-                                  commit_ref: nil,
-                                  config_map_name: nil,
-                                  service: Service.new
-                                 )
+        create_deployment
 
         hash = YAML.load(File.open(config_dir.join(filename)).read)
         value = hash.dig('spec', 'template', 'spec', 'containers', 0, 'env')
                     .find{|k,v| k['name'] == 'SENTRY_DSN' }['value']
         expect(value).to eql('runner-sentry-dsn-here')
       end
 
+      it 'sets default resources' do
+        create_deployment
+
+        hash = YAML.load(File.open(config_dir.join(filename)).read)
+        hash = hash.dig('spec', 'template', 'spec', 'containers', 0, 'resources')
+
+        expect(hash).to eql({ "limits" => { "cpu" => "150m", "memory" => "300Mi" },
+                              "requests" => { "cpu" => "10m", "memory" => "128Mi" }})
+      end
+
+      context do
```

Comment:
Just a matter of style, but if you can name this context to explain its difference. Makes it easier to understand the subtle changes from context to context.

Eg: `context "with custom resourcing but no deployment replicas set" do`

and then the next context could be `context "with only custom deployment replicas" do`.

Reply:
Yes, sorry. I would have missed this when I was refactoring out `create_deployment` and creating additional contexts